tests/Donation/TransactionSubscriberTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Tests\AppBundle\Donation;
+
+use AppBundle\DataFixtures\ORM\LoadDonationData;
+use AppBundle\Repository\DonationRepository;
+use AppBundle\Repository\TransactionRepository;
+use Tests\AppBundle\Controller\ControllerTestTrait;
+use Tests\AppBundle\MysqlWebTestCase;
+
+/**
+ * @group functional
+ */
+class TransactionSubscriberTest extends MysqlWebTestCase
+{
+    use ControllerTestTrait;
+    /**
+     * @var DonationRepository
+     */
+    private $donationRepository;
+    /**
+     * @var TransactionRepository
+     */
+    private $transactionRepository;
+
+    public function payloadProvider(): iterable
+    {
+        yield 'transaction1' => ['424241'];
+        yield 'transaction2' => ['424242'];
+        yield 'transaction3' => ['424243'];
+    }
+
+    /**
+     * @dataProvider payloadProvider
+     */
+    public function testOnPayboxIpnResponse(string $transactionId): void
+    {
+        $params = $this->createPayload($transactionId);
+        static::assertNull($this->transactionRepository->findByPayboxTransactionId($transactionId));
+
+        $this->createSignature($params);
+
+        $this->client->request('POST', '/don/payment-ipn/1528114726', $params);
+
+        static::assertSame('OK', $this->client->getResponse()->getContent());
+
+        static::assertNotNull($this->transactionRepository->findByPayboxTransactionId($transactionId));
+    }
+
+    /**
+     * @depends testOnPayboxIpnResponse
+     */
+    public function testAllTransactionCreated(): void
+    {
+        $transactions = $this->transactionRepository->findAllSuccessfulTransactionByEmail('jacques.picard@en-marche.fr');
+
+        // 4 because there are the initiale transaction when donationFixtures are loaded
```

Comment:
b/c there are initial transactions*

Reply:
There is 1 initial transaction then 1+3 = 4